src/App.js

```diff
@@ -1,30 +1,69 @@
-import React, { Fragment, useEffect } from 'react'
+import React, { useState } from 'react'
 import { BrowserRouter, Route, Switch } from 'react-router-dom'
-
+import flatten from 'flat'
 import { Provider } from 'react-redux'
-import store from './store'
+import { store, persistor } from './store'
+import { PersistGate } from 'redux-persist/integration/react'
+
 import SignInPage from './components/pages/SignInPage'
 import SignUpPage from './components/pages/SignUpPage'
 import ProtectedRoutes from './components/routes/ProtectedRoutes'
 import Home from './components/Home'
+import AppLocale from './languageProvider'
+import { IntlProvider } from 'react-intl'
+import Footer from './components/Footer/Footer'
 import './App.scss'
+import ForgotPasswordPage from './components/pages/ForgotPasswordPage'
 
 const App = () => {
+  const [currentLocale, setCurrentLocale] = useState(navigator.language)
+  const currentAppLocale =
+    currentLocale in AppLocale ? AppLocale[currentLocale] : AppLocale.en
+
+  const onChangeLocaleSelect = event => {
+    const {
+      target: { value }
+    } = event
+    setCurrentLocale(value)
+  }
+
   return (
     <Provider store={store}>
-      <BrowserRouter>
-        <Switch>
-          <ProtectedRoutes path='/' exact>
-            <Home />
-          </ProtectedRoutes>
-          <Route path='/sign-in'>
-            <SignInPage />
-          </Route>
-          <Route path='/sign-up'>
-            <SignUpPage />
-          </Route>
-        </Switch>
-      </BrowserRouter>
+      <PersistGate persistor={persistor}>
+        <IntlProvider
+          locale={currentLocale}
+          messages={flatten(currentAppLocale.messages)}
+        >
+          <BrowserRouter>
+            <Switch>
+              <ProtectedRoutes path='/' exact>
+                <Home />
+              </ProtectedRoutes>
+              <Route path='/sign-in'>
+                <SignInPage />
+                <Footer
```

Comment:
Please remove this footer as we don't need to have a select component to change the language